src/large_pages/node_large_page.cc

```diff
@@ -263,18 +263,17 @@ bool IsTransparentHugePagesEnabled() {
   // File format reference:
   // https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/mm/huge_memory.c?id=13391c60da3308ed9980de0168f74cce6c62ac1d#n163
   ifs.open("/sys/kernel/mm/transparent_hugepage/enabled");
-  if (!ifs) {
+  if (!ifs || !ifs.is_open()) {
```

Comment:
```suggestion
  if (!ifs.good()) {
```
nit: since `good` is the strictest according to the table here (**See also** section): https://en.cppreference.com/w/cpp/io/basic_ios/good

Reply:
@RaisinTen Breaking this down into two sub-changes:

1. Removal of the null-variable test: this seems worthwhile and safe, since the `std::ifstream ifs` -- a typedef'd reference to `std::basic_istream` -- object has been declared on the stack, and should be fully-constructed whenever this line of code is reached.

1. Switching from an `is_open()` to `good()` test: this took me a little while to consider, but yes, going by your reference and also the related https://en.cppreference.com/w/cpp/io/ios_base/iostate documentation, that looks fine to me.  A failed file open would set the `failbit` on the file object, causing `good()` to return `false`.  File open should be the only operation we've attempted by the time we reach this line -- so a test using `good()` seems stricter while also being correct during normal operational conditions.

Item (1) I think is definitely worth including.  Item (2) I'm less sure of - it implies that we're confident there won't be other file operations that could take place (permissions checks?) in future, and also could arguably be slightly less readable to a developer.  What do you reckon?

Reply:
1. It's an operator overload though it does look like a null-check: [https://en.cppreference.com/w/cpp/io/basic_ios/operator!](https://en.cppreference.com/w/cpp/io/basic_ios/operator!)
2. In the current situation `good` seems pretty appropriate to me because we start reading the file after the check right away. However, if you're willing to do permission checks, I agree `good` is not the best option here. Would you like to check for permissions in this PR?

Reply:
I don't know of a reason to introduce a permissions check on this configuration path from within the `node` binary; if one does emerge, it'd probably be better handled via a feature request and subsequent pull request, both separate to this one.

And wow, thanks for spotting the `not` operator overload, I didn't think to check for that.  Based on the way that overload behaves and your comment, I agree that using a call to `good()` is the way to go.